flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -493,26 +399,27 @@ public void connectSource(
      *     input-based preference.
      */
     public Collection<CompletableFuture<TaskManagerLocation>> getPreferredLocationsBasedOnInputs() {
+        final List<ConsumedPartitionGroup> allConsumedPartitions = getAllConsumedPartitions();
+
         // otherwise, base the preferred locations on the input connections
-        if (inputEdges == null) {
+        if (allConsumedPartitions == null) {
```

Comment:
Can `allConsumesPartitions` ever be `null`? If so, then I would suggest to let `getAllConsumedPartitions` rather return `Collections.emptyList()`.

Reply:
It seems that this method is going to be removed in [FLINK-21576](https://github.com/apache/flink/pull/15069).